docs/proposals/object-search.md

```diff
@@ -0,0 +1,160 @@
+![](./images/object-search-service-crd.png)
+
+### Use Cases
+
+* User can create typical multi-edge collaborative object search applications with providing AI models.
+
+## Design Details
+### CRD API Group and Version
+The `ObjectSearchService` CRD will be namespace-scoped.
+The tables below summarize the group, kind and API version details for the CRD.
+
+* ObjectSearchService
+
+| Field                 | Description             |
+|-----------------------|-------------------------|
+|Group                  | sedna.io     |
+|APIVersion             | v1alpha1                |
+|Kind                   | ObjectSearchService   |
+
+### Object search service CRD
+Below is the CustomResourceDefinition yaml for `ObjectSearchService`:
+
+[crd source](/build/crd-samples/sedna/objectsearchservice_v1alpha1.yaml)
+
+### Object search service type definition
+
+[go source](/pkg/apis/sedna/v1alpha1/objectsearchservice_types.go)
+
+#### Validation
+[Open API v3 Schema based validation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#validation) can be used to guard against bad requests.
+Invalid values for fields (example string value for a boolean field etc) can be validated using this.
+
+Here is a list of validations we need to support :
+1. The `model` specified in the crd should exist in k8s.
+2. The edge node name specified in the crd should exist in k8s.
+
+### Object search service sample
+See the [source](./crd-samples/multiedgeinferenceservice_v1alpha1.yaml) for an example.
+
+## Controller Design
+The object search service controller starts three separate goroutines called `upstream`, `downstream` and `object-search-service` controller. These are not separate controllers as such but named here for clarity.
+- object-search-service: watch the updates of object-search-service task crds, and create the workers to complete the task.
+- downstream: synchronize the object-search-service updates from the cloud to the edge node.
+- upstream: synchronize the object-search-service updates from the edge to the cloud node.
+
+### Object search service Controller
+![](./images/object-search-controller.png)
+
+The object-search-service controller watches for the updates of object-search-service tasks and the corresponding pods against the K8S API server.
+Updates are categorized below along with the possible actions:
+
+| Update Type                    | Action                                       |
+|-------------------------------|---------------------------------------------- |
+|New Object-search-service Created             |Create the cloud/edge worker|
+|Object-search-service Deleted                 | NA. These workers will be deleted by GM.|
+|The corresponding pod created/running/completed/failed                 | Update the status of object-search-service task.|
+
+
+### Downstream Controller
+![](./images/object-search-downstream-controller.png)
```

Comment:
Inconsistent: objects-search VS object-search
![image](https://user-images.githubusercontent.com/419660/126967871-95251d23-2e5d-45a1-8d0b-3f89feff3989.png)

Reply:
i prefer object not objects, proper name usually use the single, such as object detection

Reply:
All changed to "object ".